### PR TITLE
Avoid NPE if addPerfListener is called out of order with creating perfManager.getCurrentStats()

### PR DESCRIPTION
**Pull request by jacob314**
This should fix
https://github.com/flutter/flutter-intellij/issues/3173

### CODE REVIEW

**Comment by jacob314** [comment]
Hold off on reviewing. I think I found one more edge case.

**Comment by jacob314** [comment]
Implemented a more robust fix. Now listeners are disposed when the view is destroyed but not when the app is just rerun.

File: src/io/flutter/perf/FlutterWidgetPerfManager.java [hunk 1]
```diff
@@ -318,6 +324,21 @@ public void dispose() {
     if (currentStats != null) {
       currentStats.dispose();
       currentStats = null;
+      listeners.clear();
```
Comment (devoncarew) [inline src/io/flutter/perf/FlutterWidgetPerfManager.java]: Should we instead call `removePerfListener` for each listener?

Comment (jacob314) [reply src/io/flutter/perf/FlutterWidgetPerfManager.java]: I don't think we need  to as the stats object the listeners are added to is disposed a couple lines earlier.

**Pull request merged**